src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -143,6 +143,28 @@ impl ScalarFunc {
         Ok(Some(s))
     }
 
+    pub fn right<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
+        let i = try_opt!(self.children[1].eval_int(ctx, row));
+        if i <= 0 {
+            return Ok(Some(Cow::Owned(b"".to_vec())));
+        }
+        let len = s.chars().count();
```

Comment:
maybe you can write

```rust
let i = i as usize;
```

here, so that you won't need to write cast twice (which is safe because we have a `if i <= 0` right before it).